.github/workflows/test_and_build.yml

```diff
@@ -0,0 +1,65 @@
+name: Check lesson and build for all configs
+
+on:
+  pull_request:
+  push:
+    branches:
+      - gh-pages
+
+jobs:
+  spellcheck:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+    - name: Install codespell
+      run: |
+        pip3 install codespell
+    - name: Check spelling
+      run: |
+        codespell --skip="assets,*.svg,bin" --quiet-level=2  -L "rouge,dropse,namd,hist"
+
+  check_lesson_and_build_default:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-ruby@v1
+      with:
+        ruby-version: '2.7'
+    - name: Install basic requirements
+      run: |
+        # Need this library for nokogiri
+        sudo apt-get install libxslt1-dev    
+        gem install bundler jekyll json kramdown
+        bundle config set path '.vendor/bundle'
+        bundle config build.nokogiri --use-system-libraries
+    - name: "Check lesson"
+      run: |
+        make lesson-check-all
+    - name: "Check build"
+      run: |
+        make --always-make site
+
+  build-specific-sites:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        SITE_CONFIG: [ComputeCanada_Graham_slurm, EPCC_Cirrus_pbs, Norway_SIGMA2_SAGA_slurm, UCL_Myriad_sge]
```

Comment:
* Array missed an entry, `NIST_CTCMS_slurm`
* Having these all on one line is hard to parse & maintain. Please try the alternative syntax for a YAML array,
  ```yaml
  matrix:
    SITE_CONFIG:
      - ComputeCanada_Graham_slurm
      - EPCC_Cirrus_pbs
      - NIST_CTCMS_slurm
      - Norway_SIGMA2_SAGA_slurm
      - UCL_Myriad_sge
  ```

Reply:
Like so, I guess:
```suggestion
        SITE_CONFIG:
    - ComputeCanada_Graham_slurm
    - EPCC_Cirrus_pbs
    - NIST_CTCMS_slurm
    - Norway_SIGMA2_SAGA_slurm
    - UCL_Myriad_sge
```

Reply:
Looks like you need 6 more spaces at the beginning of each line.